[4.0] Multilingual status: adding more infos and clarifying some

infograf768 opened the pull request:
### Summary of Changes
This PR adds a Notice and displays the language concerned when the Content Language for that language has been deleted and remains in the site an orphan home page while the site language is published.
It also adds a Notice when a Content Language is in the trash.

It also reinstate all states of a content language to display the correct icons.
published = 1
publsihed = 0
published = -2 (i.e. trash) and for this one displays the trash icon

### Testing Instructions
Install a multilingual site with all languages available to install
French fr-FR
German de-DE
Persian fa-IR

Then use the Cpanel Multilingual Sample Data module to create a basic.

First test:
Display the Content languages Manager
Trash AND delete the fa-IR Persian content language.

You should get this when displaying the multilingual status module

<img width="1464" alt="screen shot 2019-03-02 at 18 25 09" src="https://user-images.githubusercontent.com/869724/53685264-a3e95c80-3d18-11e9-9722-8e0f1c2d1e62.png">

Then unpublish the French fr-FR Content Language.
And trash (do NOT delete) the German de-DE Content language.
I.e. Manager should look like this
<img width="671" alt="screen shot 2019-03-02 at 18 28 10" src="https://user-images.githubusercontent.com/869724/53685300-03476c80-3d19-11e9-84bc-7b05a02f92bc.png">

You should get this when displaying the multilingual status module
<img width="1486" alt="screen shot 2019-03-02 at 18 44 18" src="https://user-images.githubusercontent.com/869724/53685466-4e627f00-3d1b-11e9-9ac4-5bba6921afc8.png">

### Edit
Unpublish also the fr-FR Home page to get a new notice.
Deleted Content Language error now displays the notice as last.
see https://github.com/joomla/joomla-cms/pull/24062#issuecomment-469004768

### Reminder
It is OK to have an Unpublished Content Language (as French is above) as it lets the site owner prepare a new Language for the multilingual site.

infograf768 committed:
[4.0] Multilingual status: adding more infos and clarifying some

infograf768 committed:
Adding Notice when content language is trashed

infograf768 committed:
cs taking off older tab

Quy commented on administrator/components/com_languages/tmpl/multilangstatus/default.php:
```diff
@@ -209,11 +241,21 @@
 				<?php endif; ?>
 				<?php // Published Content languages ?>
-				<?php if ($status->lang_code && $status->published) : ?>
+				<?php if ($status->lang_code && $status->published == 1) : ?>
```
Change it similar to lines 289 - 300 where all the if statements are inside one `<td class="text-center">` and `</td>`.

infograf768 replied:
Makes sense. Will do.
Just found out some issues I have to solve also. So this is still WIP.

infograf768 committed:
Simplifying and adding a Notice

infograf768 commented:
@Quy 
Modification done.
Added new Notice when both Content Language and Home page are unpublished.

We should now get in that case:

<img width="1286" alt="screen shot 2019-03-03 at 10 22 15" src="https://user-images.githubusercontent.com/869724/53693319-24eb3700-3d9f-11e9-9c3e-28000bfcd3e9.png">

can be fully tested now.

Quy commented:
This is not working for me. The `Published Default Home pages` has all checks even though all pages are unpublished.

`Added new Notice when both Content Language and Home page are unpublished.`

Quy commented on administrator/components/com_languages/tmpl/multilangstatus/default.php:
```diff
@@ -11,9 +11,14 @@
-$notice_switchers = !$this->switchers && ($this->homes > 1 || $this->language_filter);
+$notice_homes           = $this->homes == 2 || $this->homes == 1 || $this->homes - 1 != count($this->contentlangs) && ($this->language_filter || $this->switchers != 0) && count($this->site_langs) == count($this->contentlangs);
+$notice_disabled        = !$this->language_filter && ($this->homes > 1 || $this->switchers != 0);
+$notice_switchers       = !$this->switchers && ($this->homes > 1 || $this->language_filter);
```
Remove extra spaces before `=`.

infograf768 replied:
will do

Quy replied:
Do the same for above.

infograf768 replied:
done

Quy commented on administrator/components/com_languages/tmpl/multilangstatus/default.php:
```diff
@@ -51,6 +56,34 @@
+		<?php endforeach; ?>
+		<?php // Displays error when both Content Language and Home page are unpublished ?>
+		<?php foreach ($this->statuses as $status) : ?>
+			<?php if ($status->lang_code && $status->published == 0 && !$status->home_language) : ?>
```
Can this section be combined with the above loop?

infograf768 replied:
I guess so. On it.

infograf768 commented:
>> Added new Notice when both Content Language and Home page are unpublished.

> The Published Default Home pages has all checks even though all pages are unpublished.

I do not understand this comment.

infograf768 committed:
further simplifications

Quy commented:
I don’t get the same result in your last screenshot.

infograf768 commented:
> I don’t get the same result in your last screenshot.

Please explain with more details. 
The exact settings you have for each language and the resulting screenshot.

Quy commented:
I am no longer at my computer so

infograf768 committed:
extra spaces

infograf768 commented:
I may have forgotten to ask testing by unpublishing the French Home page AND its content language, in order to get the Notice and the right icons.
The order of the Notices may also have  changed. One has to test multiple cases.

infograf768 commented:
We may need though another Notice when site language is enabled, content language is published and home page is unpublished.

I.e. this case for French and German
<img width="1368" alt="screen shot 2019-03-04 at 08 41 19" src="https://user-images.githubusercontent.com/869724/53717735-6213ff80-3e59-11e9-82a6-b89142a1c80a.png">


On it.

infograf768 committed:
add notice when both content language and home page missing. add trash
icon when home page is in the trash

infograf768 commented:
Added notice when both content language and home page missing. added trash icon when home page is in the trash.

We now get in these cases:

<img width="1466" alt="screen shot 2019-03-04 at 12 40 18" src="https://user-images.githubusercontent.com/869724/53731340-be3b4b80-3e7a-11e9-996a-d914b053d918.png">

infograf768 committed:
warning icon when home page does not exist

infograf768 commented:
Added warning icon when home page does not exist.
<img width="1505" alt="screen shot 2019-03-04 at 12 56 37" src="https://user-images.githubusercontent.com/869724/53732061-ffccf600-3e7c-11e9-9c11-af880e58f3b7.png">

infograf768 committed:
ordering new lang strings

infograf768 commented:
@alikon

alikon commented on administrator/components/com_languages/Helper/MultilangstatusHelper.php:
```diff
@@ -101,8 +101,8 @@ public static function getStatus()
-			->select('l.language AS home_language')
-			->join('LEFT', '#__menu AS l ON l.language = a.lang_code AND l.home=1 AND l.published=1 AND l.language <> \'*\'')
+			->select('l.published AS home_published')
+			->join('LEFT', '#__menu AS l ON l.language = a.lang_code AND l.home=1 AND l.language <> \'*\'')
```
please use `quoteName()`

infograf768 replied:
There are many places in this helper where we should use `quoteName()`
Shall we not reserve this for another specific patch ?

infograf768 replied:
and also `$db->quote()`

Quy commented:
Please fix conflicts.

infograf768 commented:
@Quy 
This needs a lot of work to solve the conflicts and also update the queries.
I may have to create a new PR
Will try when I have time.

infograf768 committed:
Solving conflicts. Adding quotename()

infograf768 committed:
space

infograf768 commented:
I am afraid this branch is dead. Please test this new PR
https://github.com/joomla/joomla-cms/pull/27184

Clocing
